Move discussion board to the Pattern Library

andy-armstrong opened the pull request:
## [TNL-4622](https://openedx.atlassian.net/browse/TNL-4622)

This change converts the discussion board to render using the Pattern Library. The main challenge here was in consuming the existing discussion Sass partials within the new Sass root file. 

I've also implemented some of the UX improvements from the new design where it was simpler to do so than to replicate the existing behavior. You can see the PDF design document here: https://openedx.atlassian.net/secure/attachment/33814/Discussions.pdf

Unfortunately I have a quality failure caused by touching one line in a Mako template that has some horrendous inline JavaScript. I think I'm going to clean that up in a separate PR that I'll land first.

### Sandbox
You can see the three types of discussion components here:
- Discussion board - https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/discussion/forum
- Inline discussion - https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/courseware/social_integration/48ecb924d7fe4b66a230137626bfa93e/
- Team discussion - https://andy-armstrong.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/teams/#teams/fossil-fuels/solar-power-48adbf4d8e2a4ca7b1aadcf91d2f3774

### Testing
- [ ] i18n
- [ ] RTL
- [ ] safecommit shows 0 violations
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @bjacobel 
- [ ] Code review: @dianakhuang 
- [x] UX review: @chris-mike 

FYI @AlasdairSwan @cahrens 

### Post-review
- [ ] Squash commits

andy-armstrong commented:
jenkins run js

bjacobel commented on common/static/common/templates/discussion/thread.underscore:
```diff
@@ -1,29 +1,28 @@
             <% if (can_create_comment && !readOnly) { %>
             <form class="discussion-reply-new" data-id="<%- id %>">
-                <h4><%- gettext("Post a response:") %></h4>
+                <h4><%- gettext("Add a response:") %></h4>
```
This seems like a good change to me, but should things like this be done in a separate PR? It's not really central to the effort of the pattern-library-ification.

bjacobel replied:
(That way we can have docs/product/UX review those changes on their own without the noise of infra changes going on at the same time.)

andy-armstrong replied:
That's an interesting point. I made this change because I was modifying the other buttons to fit the new design, and this label change was mentioned with those. It seemed so trivial to do that I figured I'd just include it.

@catong Will this cause you unnecessary work? Do you need to retake screenshots of the discussions because we are cleaning up the UX? Is this label mentioned in the doc strings?

catong replied:
@andy-armstrong @bjacobel Thanks for alerting me! Brian makes a good point - in general, we do need to update doc to reflect changed button labels - in procedure steps, as well as in screenshots - and Doc is not always aware of such changes if they're not done as part of some feature design work. But in this particular case I'd say go ahead - this seems to be a rare (only?) case remaining from what I remember as a previous effort by you, Andy, to standardize the verb to "Add" throughout most of the discussion area. Thanks!

bjacobel commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -6,23 +6,11 @@ body.discussion {
   .edit-post-form {
     @include clearfix();
-    @include box-sizing(border-box);
+    box-sizing: border-box;
```
:thumbsup:

clrux replied:
I think the `@include` uses Bourbon and automatically spits out different rules for various browsers.

bjacobel commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -71,7 +59,7 @@ body.discussion {
   .thread-title {
     display: block;
-    margin-bottom: $baseline;
+    margin-bottom: ($baseline);
```
Haven't seen these parens around a sass variable before, curious if there's some issue here this is meant as a workaround for?

andy-armstrong replied:
That was a mistake on my part. I had to add "+ px" to every line like this because the Pattern Library didn't include it, but then that in turn caused problems for the legacy non-Pattern Library pages. I went back and removed the suffix but then left the unnecessary parens. I've removed them now.

bjacobel commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -169,8 +157,8 @@ body.discussion {
-    background-image: url('/static/images/wmd-buttons-transparent.png');
-    background-position: 0px 0px;
+    background-image: url('#{$static-path}/images/wmd-buttons-transparent.png');
+    background-position: 0 0;
```
The default value for `background-position` is `0% 0%` so I think you might be able to lose this line completely?

andy-armstrong replied:
Yep, you're right. I tried it without this and it works fine, so I've removed it.

bjacobel commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -14,15 +14,15 @@
 
 .forum-nav-browse {
-    @include box-sizing(border-box);
+    box-sizing: border-box;
```
We're using `box-sizing: border-box` a lot, are there any block elements we *don't* wan't that set for? (ie, can we do `div { b-s: bb; }`?)

I guess that's more of a review of the original structure of the stylesheet, we should just land this first and then think about cleaning it up after.

andy-armstrong replied:
I agree with both points. There is a *lot* of duplication in these stylesheets that I'd like to clean up, but I won't do it in this PR.

bjacobel commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -81,18 +81,18 @@
 
 .forum-nav-browse-filter .icon {
-  @include font-size(12);
+  font-size: 12px;
```
I'm curious why we had to move away from the `font-size` mixin?

andy-armstrong replied:
The ``font-size`` mixin wasn't originally available so I switched a lot of the lines over. I then decided it would be simpler to put a shim mixin in just to make it work, so I can revert this if you like. I don't really understand the benefit of doing it via a mixin... do you think it was a browser support relic?

bjacobel replied:
Yeah, I don't know what the advantage of using the mixin was. As long as we do it all one way, just using the standard CSS prop is better probably IMO.

clrux replied:
We should use one of our defined size variables, or use rem/em. Also 12px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

cahrens commented on lms/djangoapps/django_comment_client/forum/views.py:
```diff
@@ -287,7 +287,9 @@ def forum_form_discussion(request, course_key):
             'category_map': course_settings["category_map"],
-            'course_settings': json.dumps(course_settings)
+            'course_settings': json.dumps(course_settings),
+            'disable_courseware_js': True,
```
The "disable_courseware_js" makes me nervous. I can't pin down exactly why, but I wanted to share my unease. :(

andy-armstrong replied:
@cahrens That's an interesting point. My thinking was that we use ``disable_courseware_js`` in Teams which also renders the discussion component, so I couldn't think why it would be a problem here. My preference would be to try it like this because it greatly reduces the weight of the page, and address any issues that come up. Does that sound okay?

cahrens commented on lms/djangoapps/django_comment_client/forum/views.py:
```diff
@@ -402,7 +404,9 @@ def single_thread(request, course_key, discussion_id, thread_id):
-            'course_settings': json.dumps(course_settings)
+            'course_settings': json.dumps(course_settings),
+            'disable_courseware_js': True,
+            'uses_pattern_library': True,
```
Is "uses_pattern_library" something you always have to specify? If there isn't already something, it would be splendid if you created a wiki page outlining how to convert an edx-platform page to the pattern library.

andy-armstrong replied:
I added a section to the "How to add a new page" wiki page a couple of weeks ago:

https://openedx.atlassian.net/wiki/display/FEDX/How+to+add+a+new+page+to+LMS+or+Studio#HowtoaddanewpagetoLMSorStudio-UsethePatternLibrary

Let me know if you think it needs more detail.

bjacobel commented on lms/static/sass/discussion/utilities/_v1-compatibility.scss:
```diff
@@ -0,0 +1,198 @@
+}
+
+%text-sr {
+    // clip has been deprecated but is still supported
```
Should probably test this in Edge, browsers that built their engines before this was deprecated will probably have it forever but who knows about ones that were built more recently...

andy-armstrong replied:
Note that I just cut-and-pasted this as-is from the non-Pattern Library Sass. Can we just assume it works? :-)

bjacobel replied:
Sure :thumbsup:

bjacobel commented on lms/static/sass/discussion/utilities/_v1-compatibility.scss:
```diff
@@ -0,0 +1,198 @@
+  @include font-size(11);
+}
+
+%t-icon9 {
```
Could use a sass map for these `%t-icon` defs, eg:

```sass
$iconMap: {1: 48, 2: 36, 3: 24}
@each $iconSize, $fontSize in $iconMap {
  %t-icon#{$iconSize} {
    @include font-size($fontSize);
  }
}
```

Looks like this pattern would work for several of the repeated definitions in this file.

bjacobel replied:
(Room for debate on this, it reduces readability/parseability to some extent but makes the file cleaner, IMO)

andy-armstrong replied:
Again this is cut-and-paste from a non-Pattern Library file, and the intention is that this compatibility file be killed before too long.

bjacobel replied:
Got it.

bjacobel commented on lms/static/sass/discussion/views/_thread.scss:
```diff
@@ -30,11 +29,25 @@ body.discussion, .discussion-module {
-    .posted-details {
-      color: $gray-d1;
+    .post-body {
+      width: flex-grid(10,12);
```
Cool, flexbox. Should make sure whoever does the a11y review of this knows we're using it here, I'm not sure if we have any other flexbox in production and if there are a11y implications.

andy-armstrong replied:
This was already used in this file, so I just followed the same approach. I dug into it a little more and it turns out this is a Bourbon function that is being deprecated, and it doesn't look like it uses flexbox in its implementation. Don't worry, I won't just start using flexbox without some careful prep work!

bjacobel replied:
Oh! I read `flex` and got excited. No worries then.

bjacobel commented on lms/static/sass/discussion/views/_thread.scss:
```diff
@@ -179,11 +174,8 @@ body.view-in-course .discussion-article {
-    border-radius: 3px;
 
-    // CASE: no courseware context or cohort visiblity rules
+    // CASE: no courseware context or cohort visibility rules
```
😛

bjacobel commented on lms/static/sass/shared-v2/_components.scss:
```diff
@@ -0,0 +1,49 @@
+            font-size: 14px;
+
+            &:hover,
+            &:focus {
```
Some inconsistency on whether we state `.foo, .bar` selectors one per line or on the same line (see [here](https://github.com/edx/edx-platform/pull/12538/files#diff-c49c4f432d08c45862665db64fe1ff6bR204) for example). I like one per line slightly more but don't particularly mind either, we should just pick one and stick to it.

andy-armstrong replied:
I prefer one-per-line too so I'll be more consistent about it.

bjacobel commented on lms/static/sass/shared-v2/_variables.scss:
```diff
@@ -0,0 +1,7 @@
+// LMS variables
+
+$lms-background-color: palette(grayscale, white);  // Why is the Pattern Library default background different?
```
Do we know the answer to this?

andy-armstrong replied:
@AlasdairSwan @chris-mike @clrux Is there a reason why we have a slight off-white background versus a pure ``#fff``? Is there a definition somewhere of the colors we should be using?

BTW, here's the PR where these Pattern Library constants were defined:

https://github.com/edx/ux-pattern-library/pull/233

clrux replied:
See the work that Brian did in the visual direction. https://drive.google.com/drive/folders/0B7jcFrBngjmkbG01ZVVzZjFVSkU

bjacobel commented on package.json:
```diff
@@ -4,7 +4,7 @@
     "backbone": "~1.3.2",
     "coffee-script": "1.6.1",
-    "edx-pattern-library": "~0.12.4",
+    "edx-pattern-library": "git+https://github.com/edx/ux-pattern-library.git#f8d8dbcb86c22450566dc6b40db0f23ea0c6fcba",
```
Should this be merged like this, or are we waiting on a UXPL release before this goes in?

andy-armstrong replied:
I was planning on waiting for a Pattern Library release. IMO we shouldn't have code on master that refers to particular commits.

andy-armstrong replied:
FYI, the commit I need is from this PR:

https://github.com/edx/ux-pattern-library/pull/329

bjacobel commented:
Awesome work @andy-armstrong, this looks like it was a lot of painstaking work. Done with my initial review, let me know if there's anything I can clarify

andy-armstrong commented:
@bjacobel @dianakhuang I've fixed the test failures now (I needed to update the screenshot baseline image used for a login test, because I tweaked the gray color used throughout the LMS). I believe this is now complete, so please re-review.

bjacobel commented:
:thumbsup: on final review

dianakhuang commented on lms/djangoapps/teams/static/teams/templates/team-profile.underscore:
```diff
@@ -4,8 +4,10 @@
-                <button type="button" class="btn new-post-btn"><span class="icon fa fa-edit new-post-icon" aria-hidden="true"></span><%- gettext("New Post") %></button>
+            <% if ( !readOnly) { %>
+                <button type="button" class="btn-neutral btn-small new-post-btn">
+                    <i class="icon fa fa-edit" aria-hidden="true"></i> <%- gettext("New Post") %>
```
I'm pretty sure this should be a `span` not an `i` tag. I think there was a PR that went through and fixed these, and you might have had a conflict.

cahrens replied:
Thanks for catching this, @dianakhuang. Yes, no more i tags! @ssemenova's PR merged yesterday (#12579), so likely this is a bad rebase.

andy-armstrong replied:
Good catch! I must have messed up the rebase. I've removed the icon now and changed the label to match all the other "Add a Post" buttons (what was "New Post").

cahrens commented on lms/templates/discussion/_discussion_module.html:
```diff
@@ -1,11 +1,14 @@
+    </a>
     % if can_create_thread:
-        <a href="#" class="new-post-btn" role="button"><span class="icon fa fa-edit new-post-icon"></span>${_("New Post")}</a>
+        <button class="new-post-btn btn-neutral btn-small">${_("Add a Post")}</button>
```
Did you mean to get rid of the icon here?

andy-armstrong replied:
Thanks for checking. The icon removal was intentional as the new design doesn't include an edit icon:

![image](https://cloud.githubusercontent.com/assets/5985072/15733808/b7bc7afe-2857-11e6-9583-5b13eb5120c7.png)

andy-armstrong replied:
Yes, this was also intentional.

cahrens commented on lms/templates/discussion/_discussion_course_navigation.html:
```diff
@@ -6,6 +7,8 @@
 <%block name="extratabs">
 % if has_permission(user, 'create_thread', course.id):
-    <li class="right"><a href="#" class="new-post-btn" role="button"><span class="icon fa fa-edit new-post-icon"></span>${_("New Post")}</a></li>
+    <li class="right">
```
Did you mean to get rid of the icon here?

andy-armstrong replied:
Yes, this was also intentional.

cahrens commented on common/static/common/templates/discussion/thread.underscore:
```diff
@@ -1,29 +1,28 @@
-                    <button class="button add-response-btn">
-                        <span class="icon fa fa-reply" aria-hidden="true"></span>
-                        <span class="add-response-btn-text"><%- gettext("Add a Response") %></span>
+                    <button class="btn-brand btn-small add-response-btn">
```
Did you mean to get rid of the icon here?

andy-armstrong replied:
Yes, this was also intentional.

dianakhuang commented on lms/static/sass/discussion/views/_create-edit-post.scss:
```diff
@@ -87,9 +88,9 @@
+    box-sizing: border-box;
     @include white-button;
-    @include font-size(14);
+    font-size: 14px;
```
Just for my own curiosity, why are we sitting the font sizes this way now?

clrux replied:
We should use one of our defined size variables, or use rem/em.

andy-armstrong replied:
@dianakhuang When I originally had these styles working in the Pattern Library the includes were not available, so I thought it was simpler to just rewrite them. Since then I've added the include to the file _compatibility-v1.scss, so I think I'll revert these changes.

@clrux I agree that the font usages need cleaning up throughout forums. This PR is just intended to get the forums styled as-is working in the Pattern Library, so I've reported a new story to do this clean up:

https://openedx.atlassian.net/browse/TNL-4708

clrux commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -52,8 +40,8 @@ body.discussion {
     .edit-post-title {
-      @include box-sizing(border-box);
-      border: 1px solid #aaa;
+      box-sizing: border-box;
```
I think the `@include` uses Bourbon and automatically spits out different rules for various browsers.

AlasdairSwan replied:
@clrux we shouldn't need the browser prefixed rules for ```box-sizing``` anymore.

clrux commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -241,20 +228,17 @@ body.discussion {
   .discussion-column {
     @include float(right);
-    @include box-sizing(border-box);
+    box-sizing: border-box;
```
I think the `@include` uses Bourbon and automatically spits out different rules for various browsers.

clrux commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -241,20 +228,17 @@ body.discussion {
     @include float(right);
-    @include box-sizing(border-box);
+    box-sizing: border-box;
+    padding-left: ($baseline/2);
```
RTL-friendify this using `@include padding-left($baseline/2);`

andy-armstrong replied:
@clrux This is useful as I wasn't aware that this was necessary for RTL. There are a large number of pre-existing usages of the direct styles through both the forums sass and all the rest of the LMS. Is this something we have to start moving to everywhere?

clrux replied:
Any styles that affect only one side (right or left) should be RTL-ified, otherwise it'll look funky when viewing right-to-left. If you come across anything that isn't RTL-friendly we should either fix it up, or make a TODO or a ticket to get it fixed up.

andy-armstrong replied:
Thanks @clrux. I've reported a new story to clean this up:

https://openedx.atlassian.net/browse/TNL-4709

clrux commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -297,8 +281,8 @@ body.discussion {
     padding: 0 ($baseline/2);
-    @include box-sizing(border-box);
-    border: 1px solid $gray-l2;
+    box-sizing: border-box;
```
I think the `@include` uses Bourbon and automatically spits out different rules for various browsers.

clrux commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -625,7 +563,7 @@ body.discussion {
     margin-bottom: $baseline;
     @include clearfix();
-    @include box-sizing(border-box);
+    box-sizing: border-box;
```
I think the `@include` uses Bourbon and automatically spits out different rules for various browsers.

clrux commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -653,9 +591,9 @@ body.discussion {
       height: 40px;
       padding: 0 ($baseline/2);
-      @include box-sizing(border-box);
+      box-sizing: border-box;
```
I think the `@include` uses Bourbon and automatically spits out different rules for various browsers.

clrux commented on lms/static/sass/discussion/_mixins.scss:
```diff
@@ -114,21 +113,20 @@
 @mixin forum-post-label($color) {
   @extend %t-weight4;
-  @include font-size(9);
+  font-size: 9px;
```
We should use one of our defined size variables, or use rem/em. Also, 9px is tiny.

clrux commented on lms/static/sass/discussion/_mixins.scss:
```diff
@@ -148,14 +146,13 @@
 @mixin forum-user-label($color) {
-  @include font-size(9);
   @extend %t-weight5;
+  font-size: 9px;
```
We should use one of our defined size variables, or use rem/em. Also, 9px is tiny.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -14,15 +14,15 @@
 .forum-nav-header {
-  @include box-sizing(border-box);
+  box-sizing: border-box;
   display: table;
```
Using `display: table` causes screenreaders to switch into table mode and read the data like a table. This should not be used for layout.

andy-armstrong replied:
Interesting. There are 40 usages of ``display: table`` and ``display: table-cell`` in the LMS's sass. We'll have to clean these up as move over to the Pattern Library.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -14,15 +14,15 @@
 .forum-nav-browse {
-    @include box-sizing(border-box);
+    box-sizing: border-box;
     display: table-cell;
```
See my note above. Using table layouts causes screenreaders to switch reading modes. If this isn't tabular data, we shouldn't be displaying it as such.

clrux replied:
Or leave this for another PR, since it seems out of scope. Can you maybe add a comment/TODO for future?

andy-armstrong replied:
I'll add a TODO for each one in the discussion sass.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -43,21 +43,21 @@
 
   .icon {
-    @include font-size(14);
+    font-size: 14px;
```
We should use one of our defined size variables, or use rem/em.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -43,21 +43,21 @@
 
 .forum-nav-browse-current {
-  @include font-size(12);
+  font-size: 12px;
```
We should use one of our defined size variables, or use rem/em. Also 12px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -66,7 +66,7 @@
 
 .forum-nav-search .icon {
-  @include font-size(12);
+  font-size: 12px;
```
We should use one of our defined size variables, or use rem/em. Also 12px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -108,7 +108,7 @@
     border-radius: 0;
-    border-bottom: 1px solid $gray-l3;
+    border-bottom: 1px solid $forum-color-border;
     padding: ($baseline/2) ($baseline/2);
```
These are the same, so we can simplify to just `padding: ($baseline/2)`

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -131,7 +131,7 @@
 
 .forum-nav-browse-menu {
-  @include font-size(14);
+  font-size: 14px;
```
We should use one of our defined size variables, or use rem/em.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -153,23 +153,23 @@
   @include clearfix();
-  @include font-size(11);
-  border-bottom: 1px solid $gray-l3;
+  font-size: 11px;
```
We should use one of our defined size variables, or use rem/em. Also 11px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -262,7 +262,7 @@
 
   .icon {
-    @include font-size(14);
+    font-size: 14px;
```
We should use one of our defined size variables, or use rem/em.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -302,7 +302,7 @@
 
 %forum-nav-thread-wrapper-2-content {
-  @include font-size(11);
+  font-size: 11px;
```
We should use one of our defined size variables, or use rem/em. Also 11px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

clrux commented on lms/static/sass/discussion/elements/_navigation.scss:
```diff
@@ -302,7 +302,7 @@
-  @include font-size(11);
+  font-size: 11px;
   display: inline-block;
   margin-right: ($baseline/4);
```
Let's RTL-friendify this using `@include margin-right($baseline/4)`

clrux commented on lms/static/sass/discussion/utilities/_developer.scss:
```diff
@@ -44,7 +44,7 @@ body.discussion {
       // alert copy
       .message {
-        @include font-size(12);
+        font-size: 12px;
```
We should use one of our defined size variables, or use rem/em. Also 12px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

clrux commented on lms/static/sass/discussion/utilities/_developer.scss:
```diff
@@ -66,7 +66,7 @@ body.discussion {
 
       .control {
-        @include font-size(14);
+        font-size: 14px;
```
We should use one of our defined size variables, or use rem/em.

clrux commented on lms/static/sass/discussion/views/_create-edit-post.scss:
```diff
@@ -115,8 +117,8 @@
+    border: 1px solid $forum-color-border;
     border-radius: 3px;
     padding: 0 $baseline/2;
     height: 40px;
```
Anyway to convert this pixel value to something relative like `em`? If a user manually increases the font-size, and if they need something that's larger than 40px, could look funky.

clrux commented on lms/static/sass/discussion/views/_create-edit-post.scss:
```diff
@@ -206,25 +208,25 @@
     @extend %ui-depth4;
     position: absolute;
     top: 40px;
     left: 0;
```
Let's RTL-friendify this with `@include left(0)`

clrux commented on lms/static/sass/discussion/views/_create-edit-post.scss:
```diff
@@ -246,7 +248,7 @@
-    border-bottom: 1px solid $gray-l3;
+    border-bottom: 1px solid $forum-color-border;
     padding: ($baseline/2);
     font-size: 14px;
```
:(

clrux commented on lms/static/sass/discussion/views/_home.scss:
```diff
@@ -3,10 +3,10 @@
 // home - layout
 .container .view-discussion-home {
-  padding: ($baseline*2) ($baseline*2) ($baseline/2);
+  padding-left: $baseline;
```
Let's RTL-friendify this with `@include padding-left($baseline)`

clrux commented on lms/static/sass/discussion/views/_home.scss:
```diff
@@ -90,7 +89,7 @@
 
     .row-item-full, .row-item {
       font-size: 12px;
```
:(

clrux commented on lms/static/sass/discussion/views/_response.scss:
```diff
@@ -50,7 +48,7 @@
     // CASE: larger username for responses 
     .username {
-      @include font-size(14);
+      font-size: 14px;
```
We should use one of our defined size variables, or use rem/em.

clrux commented on lms/static/sass/discussion/views/_response.scss:
```diff
@@ -64,40 +62,14 @@
-  @include box-sizing(border-box);
-  @include font-size(13);
+  box-sizing: border-box;
+  font-size: 13px;
```
We should use one of our defined size variables, or use rem/em. Also 13px is tiny. 16px should be our minimum. If we absolutely need smaller, 14px.

clrux commented on lms/static/sass/discussion/views/_response.scss:
```diff
@@ -129,13 +101,12 @@
+    box-sizing: border-box;
     border-radius: 2px 2px 0 0;
     background: #009fe2;
     font-size: 9px;
```
:(

clrux commented on lms/static/sass/discussion/views/_response.scss:
```diff
@@ -146,13 +117,12 @@
+    box-sizing: border-box;
     border-radius: 2px 2px 0 0;
     background: $forum-color-community-ta;
     font-size: 9px;
```
:(

clrux commented on lms/static/sass/discussion/views/_response.scss:
```diff
@@ -190,7 +159,7 @@
       padding: ($baseline/4) ($baseline/2);
-      background-color: $white;
+      background-color: $forum-color-background;
       font-size: 14px;
```
:(

clrux commented on lms/static/sass/elements/_program-card.scss:
```diff
@@ -58,7 +58,7 @@
 
   .text-section{
-    padding: 40px $baseline + px $baseline + px;
+    padding: 40px $baseline $baseline;
```
Let's RTL-friendify this with `@include padding(($baseline*2), $baseline)`

clrux commented on lms/static/sass/elements/_program-card.scss:
```diff
@@ -91,18 +91,18 @@
       background-size: 100%;
-      width: ($baseline*0.7 + px);
-      height: ($baseline*0.7 + px);
+      width: ($baseline*0.7);
```
Seems kinda funky. Might be worth using a fixed pixel value here. This is an icon after all so the size likely won't change.

clrux commented on lms/static/sass/shared-v2/_components.scss:
```diff
@@ -0,0 +1,49 @@
+// TODO: tabs should be added to the Pattern Library
+.tabs {
+    @include clearfix();
+    @extend %reset-lists;
```
Are we still following the Talbotsford property ordering method? If so, the extend should come first.

andy-armstrong replied:
@clrux I've heard of this ordering but don't know what it is. Do you have a pointer to a description of the suggested order?

clrux replied:
https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order

I was backwards: includes first, then extends. The rest of the Sass property ordering goes out to in using the box model, so, position, margin, border, padding, background, etc.

clrux commented on lms/static/sass/shared-v2/_components.scss:
```diff
@@ -0,0 +1,49 @@
+
+        &.prominent + li {
+            @include border-left(1px solid $lms-border-color);
+            padding-left: $baseline*0.75;
```
Let's RTL-friendify this with `@include padding-left($baseline*0.75)`

clrux commented on lms/static/sass/shared-v2/_components.scss:
```diff
@@ -0,0 +1,49 @@
+            border-width: 0 0 4px 0;
+            border-bottom-color: transparent;
+            color: $lms-label-color;
+            font-size: 14px;
```
We should use one of our defined size variables, or use rem/em.

clrux commented on lms/static/sass/shared-v2/_footer.scss:
```diff
@@ -79,8 +84,8 @@
           @include span(9);
-          margin-left: $baseline + px;
-          padding-left: $baseline + px;
+          margin-left: $baseline;
```
RTL

clrux commented on lms/static/sass/shared-v2/_footer.scss:
```diff
@@ -79,8 +84,8 @@
-          margin-left: $baseline + px;
-          padding-left: $baseline + px;
+          margin-left: $baseline;
+          padding-left: $baseline;
```
RTL

clrux commented on lms/static/sass/shared-v2/_header.scss:
```diff
@@ -1,96 +1,127 @@
+        box-sizing: border-box;
+        height: 74px;
+        margin: 0 auto;
+        padding: 10px 10px 0;
```
RTL

clrux commented on lms/static/sass/shared-v2/_header.scss:
```diff
@@ -1,96 +1,127 @@
+                font-size: font-size(base);
+                font-weight: font-weight(semi-bold);
+                color: palette(grayscale, dark);
+                padding: 5px 25px 23px;
```
RTL

clrux commented on lms/static/sass/shared-v2/_layouts.scss:
```diff
@@ -0,0 +1,46 @@
+    .container {
+        border: 1px solid #c8c8c8;
+        background-color: $lms-container-background-color;
+        padding: 20px;
```
I think $baseline = 20px?

clrux commented on lms/static/sass/shared-v2/_modal.scss:
```diff
@@ -26,12 +26,12 @@
-    padding-left: ($baseline/2) + px;
-    padding-right: ($baseline/2) + px;
-    padding-bottom: ($baseline/2) + px;
+    padding-left: ($baseline/2);
```
Can these be simplified into a single property shorthand? Also keep in mind RTL.

clrux commented on lms/static/sass/shared-v2/_modal.scss:
```diff
@@ -46,15 +46,11 @@
+      margin-bottom: ($baseline*1.5);
       overflow: hidden;
-      padding: 28px $baseline + px 0;
+      padding: 28px $baseline 0;
```
RTL

clrux commented on lms/static/sass/shared-v2/_modal.scss:
```diff
@@ -170,14 +166,14 @@
 
       input[type="checkbox"] {
-        margin-right: ($baseline/4) + px;
+        margin-right: ($baseline/4);
```
RTL

clrux commented on lms/static/sass/views/_teams.scss:
```diff
@@ -692,10 +697,10 @@
-    content: "";
-    position: absolute;
+  height: 100%;
+  border-left: 2px solid $gray-l4;
```
RTL

chris-mike commented:
👍  
I see no issues.

andy-armstrong commented:
@bjacobel @dianakhuang @clrux @chris-mike Thanks for the reviews and the great feedback. I've reverted the font-size changes so that they can be handled by the new story (https://openedx.atlassian.net/browse/TNL-4708), and I've squashed my commits. I'm updating my sandbox to do one last verification, and then I'm going to go ahead and merge.

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
jenkins run a11y

andy-armstrong commented:
The a11y failure here was caused by a problem on master that I picked up when rebasing. Given that, I'm going to merge this anyway if all other tests are green.

https://github.com/edx/edx-platform/pull/12634
